kernel/calls.c

```diff
@@ -272,6 +275,20 @@ void handle_interrupt(int interrupt) {
             .fault.addr = cpu->eip,
         };
         deliver_signal(current, SIGILL_, info);
+    } else if (interrupt == INT_BREAKPOINT) {
+        lock(&pids_lock);
+        send_signal(current, SIGTRAP_, (struct siginfo_) {
+            .sig = SIGTRAP_,
+            .code = SI_KERNEL_,
+        });
+        unlock(&pids_lock);
+    } else if (interrupt == INT_DEBUG) {
+        lock(&pids_lock);
+        send_signal(current, SIGTRAP_, (struct siginfo_) {
+            .sig = SIGTRAP_,
```

Comment:
Shouldn't be needed
https://github.com/ish-app/ish/blob/95dc3f5290efcd90fe009dab63cd19bcdfe32ca4/kernel/signal.c#L60

Reply:
I didn't have that at first, but IIRC this was necessary because the signal is swallowed by `signal_delivery_stop` and made available via `ptrace(PTRACE_GETSIGINFO, …)`, at which point the signal needed to be correct.